packages/@react-spectrum/dialog/src/DialogTrigger.tsx

```diff
@@ -144,13 +144,48 @@ export {_DialogTrigger as DialogTrigger};
 
 function PopoverTrigger({state, targetRef, trigger, content, hideArrow, isKeyboardDismissDisabled, ...props}) {
   let triggerRef = useRef<HTMLElement>();
-
   let overlayRef = useRef<DOMRefValue<HTMLDivElement>>();
+  let isCloserPadding = false;
+
+  // For popover triggers close to the edge, decrease the container padding allowing popovers to
+  // move closer to their trigger.
+  if (triggerRef?.current) {
+    // The popover ref is not defined here, we don't know the arrow size to factor into the below.
+    let triggerRect = triggerRef.current.getBoundingClientRect();
+    let documentRect = document.body.getBoundingClientRect();
+    // 'bottom' is the default when no value is provided
+    let [axisPlacement] = props.placement?.split(' ') || ['bottom']; // Flip and RTL don't matter
+
+    // The following is the math that gets the desired beavhior in both scales, but isn't as logical
+    // as desired. For example, would love to not multiple the default padding by 2.
```

Comment:
what does this comment mean? why do we not want to multiply the default padding by 2? why are we multiplying the default padding by 2?

Reply:
I don't remember why we are multiplying by two, it feels like we shouldn't. I tried a couple of different approaches to rework the calculations and what we have is the most accurate for both scales. The desired behavior is keeping the default padding as long as possible with the popover arrow moving to the popover edge, without going off, before being hidden.